ga4gh/gff3Parser.py

```diff
@@ -0,0 +1,321 @@
+"""
+Object representation of GFF3 data and parser for GFF3 files.
+
+See: http://www.sequenceontology.org/gff3.shtml
+"""
+
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import urllib
+import copy
+import re
+import gzip
+import bz2
+import collections
+
+GFF3_HEADER = "##gff-version 3"
+
+
+class GFF3Exception(Exception):
+    """
+    Exception associated with GFF3 data.
+    """
+    def __init__(self, message, fileName=None, lineNumber=None):
+        if fileName is not None:
+            message = "{}:{}: {}".format(fileName, lineNumber, message)
+        super(GFF3Exception, self).__init__(message)
+
+# characters forcing encode for columns
+_encodeColReStr = "\t|\n|\r|%|[\x00-\x1F]|\x7f"
```

Comment:
it's easier to read this variable name without acronyms... change 'Re' to 'Regex' ... same for the vars below